File: LeetCode #35.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int len=nums.size();
        int lt=0, rt=len-1, mid;
        
        while(lt <= rt) {
            mid = lt + (rt-lt) / 2;
            
            if(nums[mid] == target)
                return mid;
            else if(nums[mid] < target)
                lt = mid+1;
            else
                rt = mid-1;  
        }
        
        if(nums[mid] > target)
            return mid;
        
        return mid+1;
    }
};

File: LeetCode #35.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int ans = nums.length;

        int start = 0, end = nums.length - 1;

        while (start <= end) {
            int mid = (start+end)/2;

            if(nums[mid] < target)
                start = mid+1;

            else {
                ans = mid;
                end = mid-1;
            }
        }

        return ans;
    }
}